lib/phoenix/pubsub.ex

```diff
@@ -21,150 +20,39 @@ defmodule Phoenix.PubSub do
 
   """
 
-  @server Phoenix.PubSub.Server
-
-  @pg_prefix :phx
 
   @doc """
-  Creates a topic for pubsub broadcast to subscribers
-
-    * topic_name - The String name of the topic
-
-  ## Examples
-
-      iex> PubSub.create("mytopic")
-      :ok
-
+  Subscribes the pid to the pg2 group for the topic
   """
-  def create(topic_name) do
-    :ok = call {:create, group(topic_name)}
-  end
+  def subscribe(server, pid, topic),
+    do: call(server, {:subscribe, pid, topic})
 
   @doc """
-  Checks if a given topic is registered as a process group
+  Unsubscribes the pid from the pg2 group for the topic
   """
-  def exists?(topic_name) do
-    call {:exists?, group(topic_name)}
-  end
+  def unsubscribe(server, pid, topic),
+    do: call(server, {:unsubscribe, pid, topic})
 
   @doc """
-  Removes topic from process group if inactive
-
-  ## Examples
-
-      iex> PubSub.delete("mytopic")
-      :ok
-      iex> PubSub.delete("activetopic")
-      {:error, :active}
-
+  Returns lists of subscriber pids of members of pg2 group for topic
   """
-  def delete(topic_name) do
-    call {:delete, group(topic_name)}
-  end
+  def subscribers(server, topic),
```

Comment:
We can get rid of this now and call it directly in the local one. The plus side, we reduce the API required in the adapter! :D